### ISSUE TITLE
add json(s), Excel and arrow data formats support

### ISSUE THREAD

**Issue by RandomFractals** [microsoft/SandDance#154]
see [Data Preview  🈸](https://marketplace.visualstudio.com/items?itemName=RandomFractalsInc.vscode-data-preview) vscode for example of how to integration those data formats: https://dev.to/tarasnovak/vscode-data-preview-for-devs-around-the-39mn

You can use or peruse my custom [Data Manager API](https://github.com/RandomFractals/vscode-data-preview/blob/master/src/data.manager.ts) & [src/data.providers](https://github.com/RandomFractals/vscode-data-preview/tree/master/src/data.providers) folder for data loading and saving implementation details to enrich SandDance with more data source choices ...

**Comment by gramster** [comment]
I am surprised this only had one vote and it was a downvote, at least for the Arrow support, as that would clearly be a good way to share data from Jupyter implementations.

**Comment by danmarshall** [comment]
@gramster - is your Arrow data stored as a file? Wondering if there was some other mechanism from Jupyter you had in mind.

**Comment by RandomFractals** [comment]
ha! I forgot about logging this.

@danmarshall would be nice to have both, from file & pipe as described in #213

**Comment by danmarshall** [comment]
@RandomFractals - does your extension support piping?

**Comment by RandomFractals** [comment]
no, but you can call it with data file uri to open data preview similar to how I suggested you integrate vega viewer with SandDance in #153 

so, you'd just call it with:

```
commands.executeCommand('data.preview', dataFileUri)
```

**Comment by RandomFractals** [comment]
and you can check if data preview is installed via get commands:

```
// execute requested data preview command
    let viewDataCommand: string = 'vscode.open'; // default
    commands.getCommands().then(availableCommands => {
      if (availableCommands.includes(this.dataPreviewCommand)) {
        viewDataCommand = this.dataPreviewCommand;
      }
      commands.executeCommand(viewDataCommand, dataUri);
    });
```
see how I do it in vega viewer: https://github.com/RandomFractals/vscode-vega-viewer/blob/master/src/vega.preview.ts#L279

**Comment by gramster** [comment]
I was thinking of data in the Plasma object store. We had an intern prototype viewing dataframes from the Jupyter notebook in VS Code in SandDance, but that involved (IIRC) serializing the data as CSV and passing it in a URL, which clearly won't scale well. I'm wondering what we could do for large datasets (obviously writing to a file on disk is an option too, and maybe that's all we really need).

**Comment by RandomFractals** [comment]
yeah, I think to have it scale, writing to disk in raw arrow data format, rather than CSV might be a better option and than have SandDance or some other extension load a user friendly data frame/grid view.

**Comment by RandomFractals** [comment]
@danmarshall have you looked into this yet?

**Comment by danmarshall** [comment]
@RandomFractals no I haven't.

**Comment by gramster** [comment]
Sorry, I missed this. We aren’t doing anything with Arrow yet, but have
been talking about using it in the future for sharing data between kernels
in polyglot notebooks.

**Comment by RandomFractals** [comment]
yeah, @gramster: that's the one scenario where I think we are close to getting it work once you go ga ;)
